sdk/core/src/policies/transport.rs

```diff
@@ -37,6 +50,184 @@ where
 
         let response = { self.http_client.execute_request2(request) };
 
-        Ok(response.await?.into())
+        Ok(response.await?)
+    }
+
+    #[cfg(feature = "mock_transport_consume")]
```

Comment:
Can you explain why you did this as features and not just implementing `Policy` for a different struct (e.g., `MockTransportPolicy`) that we pass instead to the pipeline constructor?

Reply:
Agreed. Our other SDKs take advantage of the fact that the transport is just another policy - the terminal policy (except .NET, but close enough).

Reply:
Because of goals 1 and 3. With the feature, you can interact with both Azure _and_ the mock transport without touching a single line of code (I think this is paramount in a good testing framework). 
In other words, a dev can write code testing against the mock request/responses at will. When their code works, they can use it against Azure without having to change a single bit (thus avoiding the risk of introducing new bugs).

The rationale is: this is code that will only be used in testing (automated or not) so it makes sense to compile it only during testing (in rust we used features for this).

That said, creating a `MockTransportPolicy` as well makes sense (it's just a code shuffle). I'll do it. 👍